### PR TITLE
PAE-344 - Adding support to enable the new course seat name.

### PR DESCRIPTION
**Pull request by Jacatove**
**Description:**
This PR adds the functionality to change the course seat name in get_course_seat_name by defining the setting CUSTOM_COURSE_SEAT_NAME. Let me show you the following example:

If CUSTOM_COURSE_SEAT_NAME is defined, then the resulting course seat name will be `Seat in {CUSTOM_COURSE_SEAT_NAME} <<Course Title>> course`. Otherwise, the  normal behavior takes place.

**Ticket**: [PAE-344](https://pearsonadvance.atlassian.net/browse/PAE-344)

### CODE REVIEW

**Review by diegomillan — APPROVED** [review]

**Review by ivanvgh — APPROVED** [review]

File: ecommerce/settings/base.py [hunk 1]
```diff
@@ -806,3 +806,6 @@
 # To check government purchase restriction lists
 SDN_CHECK_API_URL ="https://api.trade.gov/gateway/v1/consolidated_screening_list/search"
 SDN_CHECK_API_KEY = "sdn search key here"
+
+# To customize the seat name
```
Comment (Squirrel18) [inline ecommerce/settings/base.py]:
```suggestion
# To customize the seat name.
```

File: ecommerce/settings/base.py [hunk 2]
```diff
@@ -806,3 +806,6 @@
...
+CUSTOM_COURSE_SEAT_NAME = ''  # Seat in {CUSTOM_COURSE_SEAT_NAME} <<Course Title>> course
```
Comment (Squirrel18) [inline ecommerce/settings/base.py]:
```suggestion
CUSTOM_COURSE_SEAT_NAME = '' # Seat in {CUSTOM_COURSE_SEAT_NAME} <<Course Title>> course.
```

File: ecommerce/settings/production.py [hunk 3]
```diff
@@ -118,3 +118,6 @@ def get_env_setting(setting):
 CORS_ALLOW_HEADERS = corsheaders_default_headers + (
     'use-jwt-cookie',
 )
+
+# To customize the seat name
```
Comment (Squirrel18) [inline ecommerce/settings/production.py]:
```suggestion
# To customize the seat name.
```

File: ecommerce/settings/production.py [hunk 4]
```diff
@@ -118,3 +118,6 @@ def get_env_setting(setting):
...
+CUSTOM_COURSE_SEAT_NAME = config_from_yaml.get('CUSTOM_COURSE_SEAT_NAME', '')
```
Comment (Squirrel18) [inline ecommerce/settings/production.py:123]: Did you check that "config_from_yaml" works as expected? @Jacatove

File: ecommerce/courses/models.py [hunk 5]
```diff
@@ -129,6 +129,14 @@ def enrollment_code_product(self):
 
     def get_course_seat_name(self, certificate_type, id_verification_required):
         """ Returns the name for a course seat. """
+        if getattr(settings, 'CUSTOM_COURSE_SEAT_NAME', ''):
+            name = 'Seat in {} {} course'.format(
```
Comment (Squirrel18) [inline ecommerce/courses/models.py]: Unnecessary variable. @Jacatove

File: ecommerce/courses/models.py [hunk 6]
```diff
@@ -129,6 +129,14 @@ def enrollment_code_product(self):
...
+                getattr(settings, 'CUSTOM_COURSE_SEAT_NAME'),
```
Comment (Squirrel18) [inline ecommerce/courses/models.py]: This is used twice, assign it to a variable. @Jacatove

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

File: ecommerce/settings/production.py (hunk 4 above)
Comment (Jacatove) [reply ecommerce/settings/production.py:123]: Yes, I checked it, but I also found out that if one configures the variable in the .yml, there is no need to define it neither in settings/production nor settings/base. So, Should I remove those lines?. @Squirrel18

File: ecommerce/courses/models.py [hunk 7]
```diff
@@ -129,6 +129,13 @@ def enrollment_code_product(self):
 
     def get_course_seat_name(self, certificate_type, id_verification_required):
         """ Returns the name for a course seat. """
+        custom_seat_name = getattr(settings, 'CUSTOM_COURSE_SEAT_NAME', '')
```
Comment (Squirrel18) [inline ecommerce/courses/models.py:132]: Add a blank line after this line, please. @Jacatove

File: ecommerce/courses/models.py [hunk 8]
```diff
@@ -129,6 +129,13 @@ def enrollment_code_product(self):
...
+        if custom_seat_name:
+            return 'Seat in {} {} course'.format(
```
Comment (Squirrel18) [inline ecommerce/courses/models.py]: Could you check if this string needs to be marked for extraction? @Jacatove

File: ecommerce/settings/production.py (hunk 4 above)
Comment (Squirrel18) [reply ecommerce/settings/production.py:123]: I don't think so, let's just leave it as it is to keep track of the change. @Jacatove

File: ecommerce/courses/models.py (hunk 8 above)
Comment (Jacatove) [reply ecommerce/courses/models.py]: It is not a translatable string. @Squirrel18

**Review by Squirrel18 — APPROVED** [review]

**Pull request merged**